### PR TITLE
Tpetra::CrsGraph: Improve exception messages on insert failure

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra @tjfulle 

## Motivation

Help with Sierra SD debugging by improving exception messages on insert failure, to show information helpful for users, like the row's capacity.

## Stakeholder Feedback

Feature requested by Sierra.

## Testing

Tested locally, Linux, GCC 4.9.3, debug build.  These are error messages.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp [hunk 1]
```diff
@@ -1987,16 +1987,24 @@ namespace Tpetra {
...
+
+    const bool insertFailed =
+      numInserted == Teuchos::OrdinalTraits<size_t>::invalid();
+    if(insertFailed) {
+      constexpr size_t ONE (1);
+      const int myRank = this->getComm()->getRank();
+      std::ostringstream os;
+      os << "On MPI Process " << myRank << ": Not enough capacity to "
+        "insert " << numInputInds
+         << " ind" << (numInputInds != ONE ? "ices" : "ex")
```
Comment (brian-kelley) [inline packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp]: You could just compare against "1U" instead of using a ONE constant, right?

Comment (brian-kelley) [reply packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp]: (you don't have to change it, I'm just curious)

Comment (mhoemmen) [reply packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp]: @brian-kelley That would fix signedness and any resulting warnings, but I prefer to compare against something known to be the same type.  Integer types have caused too much pain in the past ;-) .

**Comment by mhoemmen** [comment]
@kddevin wrote:
> For completeness (for future software quality reviews), please add to your description how you tested the changes (which platform, which tests exercise them, etc.). Having this info in each PR will make future SQ reviews easier.

Done.

**Pull request merged**